### DISCUSSION TITLE
Can you update the installation to be more noob friendly?

### DISCUSSION THREAD

**Discussion by neilyo51** [rikhuijzer/ata#32] [Q&A]
I was able to get it working. Im still learning the terminal so a lot of this is over my head. But when you said start the binary. I actually didnt know where to go with that. Google said to sudo ./<name of binaryfile> on the first run. Then i had to deconstruct the directory to toml file to edit it with the key. So i cant get the prompt screen to ask questions by typing  ./ata-x86_64-apple-darwin. However I cant just type "ata" or use "ata --help". Can you provide guidance for this? Providing my history

  242  cd Downloads
  243  ./ata-x86_64-apple-darwin
  244  sudo ./ata-x86_64-apple-darwin
  245  chmod +x ata-x86_64-apple-darwin
  246  sudo ./ata-x86_64-apple-darwin
  247  ata --help
  248  vi /Users/myUser/Library/Application Support/ata.Ask-the-Terminal-Anything-(ATA)-Project-Authors.ata/ata.toml
  249  cd /Users/myUser/Library/Application
  250  cd /Users/myUser/
  251  ls
  252  cd Library
  253  cd Appl
  254  cd Application\ Support
  255  ls
  256  vi ata.Ask-the-Terminal-Anything-(ATA)-Project-Authors.ata
  257  vi ata.Ask-the-Terminal-Anything-\(ATA\)-Project-Authors.ata
  258  cd ata.Ask-the-Terminal-Anything-\(ATA\)-Project-Authors.ata
  259  ls
  260  vi ata.toml
  261  sudo vi ata.toml
  262  ata
  263  ata --
  264  sudo ./ata-x86_64-apple-darwin
  265  exit
  266  ata --help

**Comment by rikhuijzer** [comment]
Hi! Thank you for reaching out with this question. Cool that you are learning the terminal and nice that you provided a full log. That helps a lot.

The main problem here appears to be that you cannot execute the binary. Once the binary runs, you should be able to type `ata` or `ata --help` even if there is no `ata.toml` configured. What was the error you got at the second line (243 in your numbering)? I suspect that the binary is not allowed to be executed by the system. Can you go into your `Downloads` folder (`cd Downloads`) and type `ls -ahl`? This shows whether the binary can be executed. For me, for example, it shows:
```sh
$ ls -ahl
total 13M
drwxr-xr-x   7 rik users 4.0K Feb  5 19:21 .
drwxr-xr-x   6 rik users 4.0K Feb  1 10:40 ..
-rwxr-xr-x   2 rik users 4.9M Feb  5 19:21 ata
[...]
```
The multiple `x` in front of the line tell me that the binary is allowed to be executed.

If you see less than 3 `x`es, then that might be the problem. You can fix it by typing:
```sh
$ chmod +x ata-x86_64-apple-darwin
```
This changes the file permission of the "ata-x86_64-apple-darwin" file to executable (+x).

Can you say whether this works and if it doesn't work provide an error message?

**Comment by neilyo51** [comment]
Thanks for you quick response. I wasnt really expecting a response quickly. Your app is really cool. Here are my permissions. i have all 3x's. I think at 243, the error I got was that it couldnt be executed. Which is why did the **chmod** at 245
❯ ls -ll ata-x86_64-apple-darwin
-rwxr-xr-x@ 1 myUser  staff  4541128 May 11 13:15 ata-x86_64-apple-darwin

**Comment by jazubi** [comment]
I downloaded the binary for Windows and keep getting the error below (column # varies with prompt).

_"prompt::request failed with: EOF while parsing a string at line 1 column 146"_

Strangely though, 1 out of every 30 or so prompts will work. Any ideas?
